Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -82,6 +82,10 @@ bool TcpClient::send(const char* data, uint8_t len, bool forceCloseAfterSent /*
 		stream = new MemoryDataStream();
 
 	stream->write((const uint8_t*)data, len);
+
+	if(asyncTotalLen == asyncTotalSent)
```

Comment:
just from looking at it shouldn't this be after the next line `asyncTotalLen += len;`? i'm not an expert on this bit of the code, so i'm probably wrong, but it just looks odd